lib/crypto/src/crypto.erl

```diff
@@ -47,533 +56,572 @@
+    prime192v3 |
+    prime239v1 |
+    prime239v2 |
+    prime239v3 |
+    prime256v1 |
+    secp112r1 |
+    secp112r2 |
+    secp128r1 |
+    secp128r2 |
+    secp160k1 |
+    secp160r1 |
+    secp160r2 |
+    secp192k1 |
+    secp192r1 |
+    secp224k1 |
+    secp224r1 |
+    secp256k1 |
+    secp256r1 |
+    secp384r1 |
+    secp521r1 |
+    sect113r1 |
+    sect113r2 |
+    sect131r1 |
+    sect131r2 |
+    sect163k1 |
+    sect163r1 |
+    sect163r2 |
+    sect193r1 |
+    sect193r2 |
+    sect233k1 |
+    sect233r1 |
+    sect239k1 |
+    sect283k1 |
+    sect283r1 |
+    sect409k1 |
+    sect409r1 |
+    sect571k1 |
+    sect571r1 |
+    wtls1 |
+    wtls10 |
+    wtls11 |
+    wtls12 |
+    wtls3 |
+    wtls4 |
+    wtls5 |
+    wtls6 |
+    wtls7 |
+    wtls8 |
+    wtls9.
+
+-type edwards_curve_dh() :: x25519 | x448.
+
+-type edwards_curve_ed() :: ed25519 | ed448.
 
 %%%----------------------------------------------------------------
 %%% New cipher schema
 %%%
--type cipher() :: cipher_no_iv()
-                | cipher_iv()
-                | cipher_aead() .
-
--type cipher_no_iv() :: aes_128_ecb
-                      | aes_192_ecb
-                      | aes_256_ecb
-
-                      | blowfish_ecb
-                      | des_ecb
-                      | rc4 .
-
--type cipher_iv() :: aes_128_cbc
-                   | aes_192_cbc
-                   | aes_256_cbc
-
-                   | aes_128_cfb128
-                   | aes_192_cfb128
-                   | aes_256_cfb128
-
-                   | aes_128_cfb8
-                   | aes_192_cfb8
-                   | aes_256_cfb8
-
-                   | aes_128_ctr
-                   | aes_192_ctr
-                   | aes_256_ctr
-
-                   | aes_ige256
-
-                   | blowfish_cbc
-                   | blowfish_cfb64
-                   | blowfish_ofb64
-                   | chacha20
-                   | des_ede3_cbc
-                   | des_ede3_cfb
-
-                   | des_cbc
-                   | des_cfb
-                   | rc2_cbc .
-
-
--type cipher_aead() :: aes_128_ccm
-                     | aes_192_ccm
-                     | aes_256_ccm
-
-                     | aes_128_gcm
-                     | aes_192_gcm
-                     | aes_256_gcm
-
-                     | chacha20_poly1305 .
-
+-type cipher() ::
+    cipher_no_iv() |
+    cipher_iv() |
+    cipher_aead().
+
+-type cipher_no_iv() ::
+    aes_128_ecb |
+    aes_192_ecb |
+    aes_256_ecb |
+    blowfish_ecb |
+    des_ecb |
+    rc4.
+
+-type cipher_iv() ::
+    aes_128_cbc |
+    aes_192_cbc |
+    aes_256_cbc |
+    aes_128_cfb128 |
+    aes_192_cfb128 |
+    aes_256_cfb128 |
+    aes_128_cfb8 |
+    aes_192_cfb8 |
+    aes_256_cfb8 |
+    aes_128_ctr |
+    aes_192_ctr |
+    aes_256_ctr |
+    aes_ige256 |
+    blowfish_cbc |
+    blowfish_cfb64 |
+    blowfish_ofb64 |
+    chacha20 |
+    des_ede3_cbc |
+    des_ede3_cfb |
+    des_cbc |
+    des_cfb |
+    rc2_cbc.
```

Comment:
Personally I think the original formatting is much more readable.  The grouping with empty lines emphasis the naming schema that first the numbers are varied, then the letter suffix and finally the letter prefix.  This makes the grouping look logical: All aes_*_cbc, then aes_*_ctr etc.  This has a meaning for the semantics.

And I personally also favor the bars vertically aligned in front like in grammars.

Reply:
Regarding the pipes I opened an issue, where it is easier to gather all opinions https://github.com/WhatsApp/erlfmt/issues/167

Reply:
Regarding the newlines, it is still possible to preserve this grouping by adding a comment in between.  How do you feel about that?

Reply:
A possibly empty comment is ok.  I see that is hard to automatically classify a blank line as semantic information :)

Reply:
Thank you for understanding :)